### PR TITLE
Add support for extra sendmail arguments

### LINKED ISSUE #2685: Allow sendmail options in SENDMAIL_PATH
- Gitea version (or commit ref): 1.2.0+rc3-7-g74399f3
- Git version: 2.11.0-3~bpo8+1
- Operating system: Debian Jessie Stable amd64
- Database (use `[x]`):
  - [ ] PostgreSQL
  - [x] MySQL
  - [ ] MSSQL
  - [ ] SQLite
- Can you reproduce the bug at https://try.gitea.io:
  - [ ] Yes (provide example URL)
  - [ ] No
  - [x] Not relevant

## Description

Please allow to set sendmail options like "-t -i -f webmaster@domain.tld" in option SENDMAIL_PATH. The reason is simple: Without options sendmail will use username@mailserver (e.g. git@mail.domain.tld, username = username, Gitea runs on the server) as a Return-Path in the outgoing mail. This can produce various problems.

### PR DESCRIPTION
**Pull request by cez81**
As title, fix #2685 

Extra arguments are added after the standard arguments ("-F from -i")

### CODE REVIEW

**Comment by pgaskin** [comment]
Shouldn't you tokenize the arguments in the config, lest they be passed to sendmail as one long argument?

You would need to split the arguments in the config into a slice, keeping in mind escaping and quoting. There may be a lib to do this.

File: modules/mailer/mailer.go [hunk 1]
```diff
@@ -209,6 +209,7 @@ func (s *sendmailSender) Send(from string, to []string, msg io.WriterTo) error {
 	var waitError error
 
 	args := []string{"-F", from, "-i"}
+	args = append(args, setting.MailService.SendmailArgs)
```
Comment (pgaskin) [inline modules/mailer/mailer.go]: See my comment above

**Review by pgaskin — CHANGES_REQUESTED** [review]

**Comment by cez81** [comment]
@geek1011 Oops, yes you're right :)

File: modules/mailer/mailer.go (hunk 1 above)
Comment (cez81) [reply modules/mailer/mailer.go]: Just splitting it seems to be good enough unless nothing to fancy pancy? Same is done for args to GC and health check.

File: modules/setting/setting.go [hunk 2]
```diff
@@ -1344,6 +1345,7 @@ func newMailService() {
 
 		UseSendmail:  sec.Key("USE_SENDMAIL").MustBool(),
 		SendmailPath: sec.Key("SENDMAIL_PATH").MustString("sendmail"),
+		SendmailArgs: sec.Key("SENDMAIL_ARGS").Strings(" "),
```
Comment (pgaskin) [inline modules/setting/setting.go]: @cez81 this does not handle quoted or escaped strings I think.

**Review by pgaskin — CHANGES_REQUESTED** [review]

File: modules/setting/setting.go (hunk 2 above)
Comment (cez81) [reply modules/setting/setting.go]: @geek1011 New commits to use go-shellquote library

File: modules/setting/setting.go [hunk 3]
```diff
@@ -1372,6 +1374,11 @@ func newMailService() {
 	MailService.FromName = parsed.Name
 	MailService.FromEmail = parsed.Address
 
+	MailService.SendmailArgs, err = shellquote.Split(sec.Key("SENDMAIL_ARGS").String())
```
Comment (lafriks) [inline modules/setting/setting.go]: Might be it's worth load this (and also parse ) only if `MailService.UseSendmail` is enabled?

Comment (cez81) [reply modules/setting/setting.go]: Done

**Review by sapk — APPROVED** [review]

**Comment by sapk** [comment]
LGTM

**Comment by lafriks** [comment]
LGTM

**Comment by lafriks** [comment]
@geek1011 need your approval

**Review by pgaskin — APPROVED** [review]
LGTM

**Comment by lafriks** [comment]
Make LG-TM work again

**Comment by msebald** [comment]
Great, thank you. How do I use it in the config? If I put the following there is no difference when sending out the test mail:

SENDMAIL_ARGS = -t -i -f webmaster@domain.tld

**Comment by cez81** [comment]
Something like that should work. You could try -X /tmp/sendmail.log and see what goes wrong. -F and -i are set by default in the code.

**Pull request merged**